src/loaders/lottie/tvgLottieBuilder.cpp

```diff
@@ -1075,13 +1076,13 @@ void LottieBuilder::updateText(LottieLayer* layer, float frameNo)
                     auto start = nearbyintf((*s)->start(frameNo) / divisor) + offset;
                     auto end = nearbyintf((*s)->end(frameNo) / divisor) + offset;
 
-                    if (start > end) std::swap(start, end);
+                    if (start > end && end > 0) std::swap(start, end);
```

Comment:
What do you think about initializing `end` with the maximum value if it is open to expansion by default?
```
LottieFloat end = FLT_MAX;
```